### PR TITLE
[EJBCLIENT-505] CI Add integration testing to CI script

### PR DESCRIPTION
**Pull request by tadamski**
https://issues.redhat.com/browse/EJBCLIENT-505

### CODE REVIEW

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      matrix:
+        os: [ubuntu-latest]
+        jdk-distribution: [temurin]
+        jdk-version: [17]
+    steps:
+      - name: Configure runner - Linux
+        if: contains(matrix.os, 'ubuntu')
+        run: |
+          sudo bash -c "echo '127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4' > /etc/hosts"
+          sudo bash -c "echo '::1         localhost localhost.localdomain localhost6 localhost6.localdomain6' >> /etc/hosts"
```
Comment (chengfang) [inline .github/workflows/integration.yml:22]: I believe these lines are no longer needed.  They were already deleted from other workflow/*.yml files.

**Comment by chengfang** [comment]
In the integration test log, there is one line indicating that the integration testsuite is cloning jboss-ejb-client:

Executing command git clone -b main --single-branch https://github.com/wildfly/jboss-ejb-client in directory /home/runner/work/jboss-ejb-client/jboss-ejb-client/ejb-client-testsuite/snapshots/target

So the main branch is cloned, which does not contain the changes in the PR that was triggering this test run yet. So looks like the integration testsuite run will always be a step behind.

**Comment by tadamski** [comment]
@chengfang  thanks for the review - I have fixed all the mentioned issues; I have also update the build script so that the testsuite is build in two steps; this is required after merging Marek's bom PR

File: .github/workflows/integration.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+echo "INTEGRATION TESTS"
+
+EJB_CLIENT_REPOSITORY=$1
+EJB_CLIENT_BRANCH=$2
+
+git clone --depth=1 https://github.com/wildfly/ejb-client-testsuite
+
+cd ejb-client-testsuite
+
+mvn -B -ntp package -DspecificModule=prepare --batch-mode -Dejb.client.repository=https://github.com/${EJB_CLIENT_REPOSITORY} -Dejb.client.branch=${EJB_CLIENT_BRANCH}
```
Comment (chengfang) [inline .github/workflows/integration.sh:10]: --batch-mode is the same as -B, which is already in this command.

File: .github/workflows/integration.sh [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+mvn -B -ntp dependency:tree clean verify --fail-at-end --batch-mode
```
Comment (chengfang) [inline .github/workflows/integration.sh:11]: we should be able to remove --batch-mode

**Comment by chengfang** [comment]
the integration test job failed, with errors like the below (there may be other errors):

Executing command git clone -b EJBCLIENT-505  --depth=1 --single-branch https://github.com/wildfly/jboss-ejb-client jboss-ejb-client in directory /home/runner/work/jboss-ejb-client/jboss-ejb-client/ejb-client-testsuite/snapshots/target
Cloning into 'jboss-ejb-client'...
warning: Could not find remote branch EJBCLIENT-505 to clone.
fatal: Remote branch EJBCLIENT-505 not found in upstream origin

**Pull request closed without merging**